evap/evaluation/models.py

```diff
@@ -443,6 +443,38 @@ class State:
         verbose_name=_("wait for grade upload before publishing"), default=True
     )
 
+    @property
+    def has_exam(self):
+        return self.course.evaluations.filter(name_de="Klausur", name_en="Exam").exists()
+
+    def make_exam_evaluation(
+        self,
+        exam_date: datetime,
+        evaluation_end_date: datetime,
+        participants: query.QuerySet["UserProfile"],
+        eval_contributions: query.QuerySet["Contribution"],
```

Comment:
why the parameters `participants` and `eval_contributions`? I'd say they are functionally dependent on `exam_evaluation` and defining how they are deduced makes more sense inside this function than in the view (where the argument is currently calculated).